### PR TITLE
docs: Adding category view  to New Docs Site [JOB-110256]

### PR DESCRIPTION
**Pull request by Beccaneer**
### Changed

- Added category view to collection pages
- Removed old h2 from BodyBlock

## Testing

Run the docs site from the packages/site directory with `npm run dev`.

Pages that have content with a 'sections' parameter in their objects should default to the category view, and display a segmented control in the top allowing the user to flip between views. (Ex. /components)

Pages that do not have categories should just display the regular card view and not display the segmented control (Ex. /design)

Can do a scroll through and make sure things look okay and card links still work :)

Screenshot 2024-12-13 at 12 13 08 PM
Screenshot 2024-12-13 at 12 13 01 PM
Screenshot 2024-12-13 at 11 52 36 AM
Screenshot 2024-12-13 at 11 52 31 AM

Changes can be
[tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Review by chris-at-jobber — DISMISSED** [review]
Everything in the collection itself looks great! Can we keep the header block's children matched to the width of the BodyBlock? (or followup in a separate PR?)

image
image

**Comment by Beccaneer** [comment]
> Everything in the collection itself looks great! Can we keep the header block's children matched to the width of the BodyBlock? (or followup in a separate PR?)
> 
> image image

Yeah for sure! Let's put that in a separate ticket if that's okay

**Comment by Beccaneer** [comment]
@chris-at-jobber https://jobber.atlassian.net/browse/JOB-111903

**Review by chris-at-jobber — COMMENTED** [review]
Design approved, other changes happening in a followup

File: packages/site/src/components/BodyBlock.tsx [hunk 1]
```diff
@@ -1,32 +1,21 @@
...
-}
-
 /**
  * A simple opinionated wrapper for a block of content within a PageBlock
- * @param param0 {children:ReactNode, title: string}
+ * @param param0 {children:ReactNode}
  * @returns ReactNode
  */
-export const BodyBlock = ({ children, title }: BodyBlockProps) => {
+export const BodyBlock = ({ children }: PropsWithChildren) => {
```
Comment (chris-at-jobber) [inline packages/site/src/components/BodyBlock.tsx:9]: I _think_ removing the title here is what's making the site build CI step fail

Comment (Beccaneer) [reply packages/site/src/components/BodyBlock.tsx:9]: Ohhhhh yup, had to remove the body title from a couple of other places as well

File: packages/site/src/components/CategoryCardSection.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ */
+export const CategoryCardSection = ({
+  category,
+  children,
+}: PropsWithChildren<CategoryCardSectionProps>) => {
+  return (
+    <div
+      style={{
+        display: "grid",
+        gridTemplateColumns: "96px 1fr",
```
[the remark this answers is not in the document]
Comment (taylorvnoj) [reply packages/site/src/components/CategoryCardSection.tsx:24]: Ohhh this is super neat!

File: packages/site/src/components/PageBlock.tsx [hunk 3]
```diff
@@ -27,15 +30,78 @@ interface PageBlockProps {
...
+      <BodyBlock>
+        {structure.useCategories && (
+          <div
+            style={{
+              width: 305,
+              margin: "auto",
+              paddingBottom: "var(--space-large)",
+            }}
+          >
+            <SegmentedControl
```
Comment (taylorvnoj) [inline packages/site/src/components/PageBlock.tsx:73]: ❤️

**Review by taylorvnoj — APPROVED** [review]
LGTM! So great to see changes addressing early feedback already! Well done!

**Pull request merged**